front/src/applications/stdcmV2/components/StdcmTableResults.tsx

```diff
@@ -0,0 +1,188 @@
+                  <td className="stop">{isLastStep || step.duration > 0 ? step.time : ''}</td>
+                  <td className="stop">
+                    <div
+                      className={
+                        step.duration !== 0 && !isLastStep ? 'stop-with-duration ml-n2' : 'stop'
+                      }
+                      style={{
+                        width: `${step.duration < 600 && step.duration >= 60 ? 55 : 65}px`,
+                      }}
+                    >
+                      {
+                        // eslint-disable-next-line no-nested-ternary
+                        !isFirstStep && !isLastStep
+                          ? step.duration !== 0
+                            ? getStopDurationTime(step.duration)
+                            : step.time
+                          : ''
+                      }
+                    </div>
+                  </td>
+                  <td className="stop">
+                    {isFirstStep || step.duration > 0 ? step.stopEndTime : ''}
+                  </td>
+                  <td className="weight" style={{ color: isLastStep ? '#797671' : '#312E2B' }}>
+                    {!isFirstStep && !isLastStep
+                      ? '='
+                      : `${Math.floor(rollingStockData.mass / 1000)} t`}
+                  </td>
+                  <td style={{ color: isLastStep ? '#797671' : '#312E2B' }}>
+                    {!isFirstStep && !isLastStep ? '=' : rollingStockData.metadata?.reference}
+                  </td>
+                </tr>
+              );
+            }
+            return null;
+          })}
+        </tbody>
+      </table>
+      <div className="display-all">
+        <div className="button-display-all-PR">
+          <Button
+            variant="Normal"
+            label={showAllOP ? t('displayMain') : t('displayAll')}
+            onClick={handleShowAllClick}
+          />
+        </div>
+        <div className="button-get-simulation">
+          {!isSimulationSelected ? (
+            <Button
+              label={t('selectThisSimulation')}
+              onClick={selectSimulation}
+              isDisabled={isSimulationSelected}
+            />
+          ) : (
+            <div className="selected-simulation">{t('simulationSelected')}</div>
+          )}
+        </div>
+      </div>
+      {isSimulationSelected && (
+        <div className="get-simulation">
+          <div className="download-simulation">
+            <PDFDownloadLink
+              document={
+                <SimulationReportSheetV2
+                  stdcmData={stdcmData}
+                  pathProperties={pathProperties}
+                  rollingStockData={rollingStockData}
+                  speedLimitByTag={speedLimitByTag}
+                  simulationReportSheetNumber={simulationReportSheetNumber}
+                  mapCanvas={mapCanvas}
+                  creationDate={creationDate}
+                />
```

Comment:
The OpList is also used in the reportSheet, you should give it as a props to avoid computing it twice
```suggestion
              <SimulationReportSheetV2
                  stdcmData={stdcmData}
                  speedLimitByTag={speedLimitByTag}
                  simulationReportSheetNumber={simulationReportSheetNumber}
                  mapCanvas={mapCanvas}
                  opList={opList}
                />
```

Reply:
fix